src/app/components/ThemeProvider/index.tsx

```diff
@@ -1,34 +1,76 @@
 import React, { PropsWithChildren } from 'react';
+import type { LoadableComponent } from '@loadable/component';
+import nodeLogger from '#lib/logger.node';
+import { THEME_PROVIDER_ERROR } from '#app/lib/logger.const';
 import defaultServiceVariants from '../../lib/config/services/defaultServiceVariants';
-import { Services, Variants } from '../../models/types/global';
+import {
+  ServicesVariantsProps,
+  Variants,
+  ServicesWithVariants,
+  ServicesWithNoVariants,
+} from '../../models/types/global';
 import themes from './themes/loadableConfig';
 import fallBackTheme from './themes/news';
 
-interface Props {
-  service: Services;
-  variant?: Variants | null;
-}
+const logger = nodeLogger(__filename);
+
+type ThemeComponentLoadable = LoadableComponent<{ children: React.ReactNode }>;
+type FallbackThemeComponent = React.FC<{ children: React.ReactNode }>;
+
+type NonVariantThemesType = {
+  [_service in ServicesWithNoVariants['service']]: ThemeComponentLoadable;
+};
+
+type VariantThemesType = {
+  [_service in ServicesWithVariants['service']]: {
+    [_variant in Variants]?: ThemeComponentLoadable;
+  };
+};
+
+const nonVariantThemes = Object.fromEntries(
+  Object.entries(themes).filter(([_service, theme]) =>
+    Object.keys(theme).includes('render'),
+  ),
+) as unknown as NonVariantThemesType;
+
+const variantThemes = Object.fromEntries(
+  Object.entries(themes).filter(
+    ([_service, theme]) => !Object.keys(theme).includes('render'),
+  ),
+) as unknown as VariantThemesType;
 
 export const ThemeProvider = ({
   children,
   service,
-  variant,
-}: PropsWithChildren<Props>) => {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  let LoadableContextProvider: any = fallBackTheme;
+  ...rest
+}: PropsWithChildren<ServicesVariantsProps>) => {
+  let LoadableContextProvider:
+    | ThemeComponentLoadable
+    | FallbackThemeComponent
+    | null = null;
 
-  const serviceVariant: Variants = variant || defaultServiceVariants[service];
+  const variant: Variants = rest.variant || defaultServiceVariants[service];
 
-  if (serviceVariant === 'default' || !serviceVariant) {
-    LoadableContextProvider = themes[service];
+  let serviceVariants;
+  let serviceNoVariants;
+
+  if (service in variantThemes) {
+    serviceVariants =
+      variantThemes[service as ServicesWithVariants['service']][variant];
   } else {
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore - TODO: come back to this
-    LoadableContextProvider = themes[service][serviceVariant];
+    serviceNoVariants =
+      nonVariantThemes[service as ServicesWithNoVariants['service']];
   }
 
-  if (!LoadableContextProvider) {
-    return null;
+  if (serviceNoVariants || serviceVariants) {
+    LoadableContextProvider =
+      serviceNoVariants || serviceVariants || fallBackTheme;
```

Comment:
```suggestion
      serviceNoVariantTheme || serviceVariantTheme;
```
I'm not sure the fallBackTheme can ever resolve here?

Reply:
Whilst that is true, without it, TS fails, so I had to leave it as is. 

This is because I didn't want to set the fallbackTheme upfront (line 50), I only wanted to set it as a last resort, so that I could also add the log message below.

This was a bit of a headscratcher - my changes had caused an issue with the NextJS app - it was rendering a white screen, which is why we needed the fallback theme & also the error message in case it ever happens again (quite unlikely but you never know). See this [Slack thread](https://bbc-tpg.slack.com/archives/G03T00R76KX/p1760343076410309?thread_ts=1760106987.555839&cid=G03T00R76KX) for context